nvflare/app_opt/xgboost/histogram_based_v2/adaptor_executor.py

```diff
@@ -168,3 +178,31 @@ def _notify_client_done(self, rc, fl_ctx: FLContext):
             fl_ctx=fl_ctx,
             optional=True,
         )
+
+    def _get_sender(self, fl_ctx: FLContext) -> Sender:
+        """Get request sender to be used by this executor.
+
+        Args:
+            fl_ctx: the FL context
+
+        Returns:
+            A sender object
+        """
+
+        if self.sender_id:
+            engine = fl_ctx.get_engine()
+            sender = engine.get_component(self.sender_id)
+            if not sender:
+                self.system_panic(f"cannot get component for {self.sender_id}", fl_ctx)
+            else:
+                if not isinstance(sender, Sender):
+                    self.system_panic(
+                        f"invalid component '{self.sender_id}': expect {Sender.__name__} but got {type(sender)}",
+                        fl_ctx,
+                    )
+                    sender = None
+
+        else:
+            sender = SimpleSender()
```

Comment:
One question, if we have ReliableSender, why not just make it default?
Do we need this SimpleSender at all?

Reply:
ReliableSender requires additional config, so it's hard to make it default.

Reply:
Yep.